[docs] table improvements (#4730)
ToolJet/ToolJet: docs/docs/widgets/table.md

- **does not contain**
- **matches**
- **does not match**
- **equals**
- **does not equal to**
- **is empty**
- **is not empty**
- **greater than**
- **greater than or equal to**
- **less than**
- **less than or equal to**

### Show bulk update actions

It's enabled by default. Table footer will show two bulk update actions **Save changes** & **Discard changes** whenever a cell is edited. Toggle `off` to hide bulk update actions.

### Bulk selection

To let the user select one or more rows from the current page of a table, enable 'Bulk selection' from the inspector. The values of selected rows will be exposed as `selectedRows`.

### Highlight selected row

Enable this option to have the last selected(clicked on) row to be highlighted.

### Disable sorting

Enable this option to lock the sorting of columns when clicked on column name.

### Server-side sort
When Server-side sort is enabled, clicking on the column headers will not automatically sort the table, instead, the `Sort applied` event will be fired and the applied sorting will be exposed as `sortApplied`. You can use this data to run any query that feeds data to the table in a manner that reflects the sorting applied.

> ### Server-side filter
> 
> 

Author: ```suggestion
### Server-side filter
When Server-side filter is enabled, applying filters will not automatically filter the table, instead, the `Filter changed` event will be fired and the applied filters will be exposed as `filters`. You can use this data to run any query that feeds data to the table in a manner that reflects the filters applied.
```

Show or hide download button at the Table footer.

### Hide/Show columns

Table header has an option(Eye icon) to show/hide one or many columns on the table. 

### Show filter button

Show or hide filter button at the Table header. The following filters are available:
- **contains**
- **does not contain**
- **matches**
- **does not match**
- **equals**
- **does not equal to**
- **is empty**
- **is not empty**
- **greater than**
- **greater than or equal to**
- **less than**
- **less than or equal to**

### Show bulk update actions

It's enabled by default. Table footer will show two bulk update actions **Save changes** & **Discard changes** whenever a cell is edited. Toggle `off` to hide bulk update actions.

### Bulk selection

To let the user select one or more rows from the current page of a table, enable 'Bulk selection' from the inspector. The values of selected rows will be exposed as `selectedRows`.

### Highlight selected row

Enable this option to have the last selected(clicked on) row to be highlighted.

### Disable sorting

Enable this option to lock the sorting of columns when clicked on column name.

> ### Server-side sort
> 
> 

Author: ```suggestion
### Server-side sort
When Server-side sort is enabled, clicking on the column headers will not automatically sort the table, instead, the `Sort applied` event will be fired and the applied sorting will be exposed as `sortApplied`. You can use this data to run any query that feeds data to the table in a manner that reflects the sorting applied.
```

### Save changes

If any cell of the table is edited, the `save changes` button appears at the footer of the table. Save changes event is triggered when this button is clicked.

### Cancel changes

If any cell of the table is edited, the `Discard changes` button appears at the footer of the table. Cancel changes event is triggered when this button is clicked.

### Page changed

If server-side pagination is enabled, this event is fired when the current page is changed. This event is triggered after updating the `pageIndex` variable.

### Search

This event is triggered when a text is entered to the search input box of the table. `searchText` variable is updated before triggering this event.

### Sort applied

This event is triggered when the column name header is clicked to apply sorting in `asc` or `desc`. The `sortApplied` variable is updated with an object having `column` and `direction` values.

### Cell value changed

If any cell of the table is edited, the `cell value changed` event is triggered.

### Filter changed

> This event is triggered when filter is added from the header of the table. `filters` property of the table is updated with an object for each filter added. The objects will have `condition`, `value`, and `column` keys. 

Reviewer: ```suggestion
This event is triggered when filter is added, removed, or updated from the filter section of the table. `filters` property of the table is updated to reflect the status of filters applied. The objects will have properties: `condition`, `value`, and `column`. 
```
